src/routes/vendor.tsx

```diff
@@ -36,6 +38,29 @@ export default function VendorPage() {
     setShowLoadingOverlay(!isSuccess)
   }, [isSuccess, setShowLoadingOverlay])
 
+  const [vendorIds, setVendorIds] = useState<Set<string>>(new Set())
+  const handleUpdateChosenVendor = (checked: CheckedState, vendorId: string) => {
+    const udpatedSet = new Set(vendorIds)
+    if (checked)
+      udpatedSet.add(vendorId)
+    else
+      udpatedSet.delete(vendorId)
+    setVendorIds(udpatedSet)
+    // setVendorIds([...vendorIds.filter(id => id !== vendorId)])
+  }
+  const chooseAllVendor = (vendors: Vendor[], toggle: boolean) => {
+    const updatedSet = new Set(vendorIds)
+    if (toggle)
+      vendors.map(vendor => updatedSet.add(vendor.id))
+    else
+      vendors.map(vendor => updatedSet.delete(vendor.id))
+    setVendorIds(updatedSet)
+    // const updatedSet = new Set()
+    // if (vendorIds.)
+    // const notDuplicate = [vendorIds.filter(id=>vendor.)]
+    // setVendorIds([...vendorIds, ...vendors.map(vendor => vendor.id)])
+  }
```

Comment:
why do we have to use a set here? can you explain?

Reply:
Hwemmm before I'm using the basic array to store all the vendor selected from checking checkboxes right, but then implementing the checkbox on the table header popping the needs to filter, which was the vendors that has already been selected, and adding the new vendors only. event though the complexity difference will be very small (considering the amount of vendor shown only 10) but it will be optimized using Set (implemented as hashSet in JS if I'm not mistaken) which already has the method of .has(). The difference will be O(N) and O(N^2) on the filtering side and after comparing the readibility code, it serves much simpler way.

Reply:
I think a simple for loop will do, we can check from the state of each row's checkbox instead, wdyt?

Reply:
Actually good point, but it requires the usage of caching after getting responses from backend to keep data on previous page (when we navigate to the next page). Currently I'm not sure due to the cache tags that is possible to be invalidated while doing data manipulation.

Reply:
It's fine, I don't think the performance is all that difference either way, so let's just leave it as is